### PR TITLE
Add 7TV Emotes plugin

### PR DESCRIPTION
**Pull request by WaterBoiledPizza**
Made another plugin for searching emotes on 7tv. This project is based on the featured plugin [Garulf's BetterTTV-Twitch-Emotes](https://github.com/Garulf/BetterTTV-Twitch-Emotes)

### CODE REVIEW

File: plugins.json [hunk 1]
```diff
@@ -1354,5 +1354,20 @@
...
+        "Author": "Water Boiled Pizza",
+        "Version": "v0.0.1",
+        "Language": "python",
+        "Website": "https://github.com/WaterBoiledPizza/7TV-Emotes",
+        "IcoPath": "https://cdn.jsdelivr.net/gh/WaterBoiledPizza/7TV-Emotes@main/icon.png?raw=true",
+        "UrlSourceCode": "https://github.com/WaterBoiledPizza/7TV-Emotes",
+        "UrlDownload": "https://github.com/WaterBoiledPizza/7TV-Emotes/releases/download/v0.0.1/Flow.Launcher.Plugin.7tvEmotes.zip",
+        "Tested": true,
+        "DateAdded": "2023-03-11T18:53:00Z",
+        "LatestReleaseDate": "2023-03-11T18:53:00Z"
```
Comment (Garulf) [inline plugins.json:1371]:
```suggestion
```

This will be added by the CI

File: plugins.json [hunk 2]
```diff
@@ -1354,5 +1354,17 @@
...
+        "ID": "85eb90ff-b92d-476b-bae4-3d71398e61f9",
+        "Name": "7TV Emotes",
+        "Description": "Search emotes from 7TV",
+        "Author": "Water Boiled Pizza",
+        "Version": "v0.0.1",
+        "Language": "python",
+        "Website": "https://github.com/WaterBoiledPizza/7TV-Emotes",
+        "IcoPath": "https://cdn.jsdelivr.net/gh/WaterBoiledPizza/7TV-Emotes@main/icon.png?raw=true",
+        "UrlSourceCode": "https://github.com/WaterBoiledPizza/7TV-Emotes",
+        "UrlDownload": "https://github.com/WaterBoiledPizza/7TV-Emotes/releases/download/v0.0.1/Flow.Launcher.Plugin.7tvEmotes.zip",
```
Comment (Garulf) [inline plugins.json:1368]:
```suggestion
        "UrlDownload": "https://github.com/WaterBoiledPizza/7TV-Emotes/releases/download/v0.0.1/Flow.Launcher.Plugin.7tvEmotes.zip"
```

Remove extra `,`.

**Review by Garulf — CHANGES_REQUESTED** [review]
Looks like the plugin raised an exception during the run test:

```

{"result": [{"Title": "TypeError", "SubTitle": "'NoneType' object is not subscriptable", "IcoPath": "C:\\Users\\runneradmin\\AppData\\Local\\FlowLauncher\\Images\\app_error.png", "ContextData": null, "Score": 0, "JsonRPCAction": {"method": "change_query", "parameters": [], "dontHideAfterAction": true}, "AutoCompleteText": "7tv TypeError"}, {"Title": "Report Issue", "SubTitle": "Report this issue to the developer", "IcoPath": "C:\\Users\\runneradmin\\AppData\\Local\\FlowLauncher\\Images\\browser.png", "ContextData": null, "Score": 0, "JsonRPCAction": {"method": "create_github_issue", "parameters": ["TypeError", "Traceback (most recent call last):%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\lib\\flox\\launcher.py\", line 44, in run%0A    results = request_method(*request_parameters) or self._results%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\lib\\flox\\__init__.py\", line 132, in _query%0A    self.query(query)%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\plugin\\main.py\", line 14, in query%0A    emotes = cache(seventv.trending_emotes)()%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\lib\\flox\\utils.py\", line 41, in wrapper%0A    data = func(*args, **kwargs)%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\plugin\\seventv.py\", line 65, in trending_emotes%0A    return _request(category=\"TRENDING_DAY\")%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\plugin\\seventv.py\", line 54, in _request%0A    data = data['data']['emotes']['items']%0ATypeError: 'NoneType' object is not subscriptable%0A"], "dontHideAfterAction": false}, "AutoCompleteText": "7tv Report Issue"}]}
```

Are you able to look into this?

**Comment by WaterBoiledPizza** [comment]
> Looks like the plugin raised an exception during the run test:
> 
> ```
> 
> {"result": [{"Title": "TypeError", "SubTitle": "'NoneType' object is not subscriptable", "IcoPath": "C:\\Users\\runneradmin\\AppData\\Local\\FlowLauncher\\Images\\app_error.png", "ContextData": null, "Score": 0, "JsonRPCAction": {"method": "change_query", "parameters": [], "dontHideAfterAction": true}, "AutoCompleteText": "7tv TypeError"}, {"Title": "Report Issue", "SubTitle": "Report this issue to the developer", "IcoPath": "C:\\Users\\runneradmin\\AppData\\Local\\FlowLauncher\\Images\\browser.png", "ContextData": null, "Score": 0, "JsonRPCAction": {"method": "create_github_issue", "parameters": ["TypeError", "Traceback (most recent call last):%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\lib\\flox\\launcher.py\", line 44, in run%0A    results = request_method(*request_parameters) or self._results%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\lib\\flox\\__init__.py\", line 132, in _query%0A    self.query(query)%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\plugin\\main.py\", line 14, in query%0A    emotes = cache(seventv.trending_emotes)()%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\lib\\flox\\utils.py\", line 41, in wrapper%0A    data = func(*args, **kwargs)%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\plugin\\seventv.py\", line 65, in trending_emotes%0A    return _request(category=\"TRENDING_DAY\")%0A  File \"C:\\Users\\runneradmin\\AppData\\Roaming\\FlowLauncher\\Plugins\\7TV Emotes\\plugin\\seventv.py\", line 54, in _request%0A    data = data['data']['emotes']['items']%0ATypeError: 'NoneType' object is not subscriptable%0A"], "dontHideAfterAction": false}, "AutoCompleteText": "7tv Report Issue"}]}
> ```
> 
> Are you able to look into this?

what was the test input?

**Comment by WaterBoiledPizza** [comment]
also where can i see the error logs when running the plugin? Is it plugin.log?

**Comment by Garulf** [comment]
I see the issue. It's one I've struggled with myself. 

I'll elaborate once I'm at my computer again soon.

**Comment by Garulf** [comment]
> what was the test input?

It's sent an empty query: `""`

> also where can i see the error logs when running the plugin? Is it plugin.log?

Yes it should output there.

**Comment by WaterBoiledPizza** [comment]
> It's sent an empty query: `""`

I see the problem when i tried with the empty characters. Either way i can put a RegEX to filter out the excluded characters.

**Comment by Garulf** [comment]
> > It's sent an empty query: `""`
> 
> I see the problem when i tried with the empty characters. Either way i can put a RegEX to filter out the excluded characters.

Okay. Feel free to update the PR with your latest release when you want.

**Review by Garulf — APPROVED** [review]

**Pull request merged**